tdms/include/interpolation_methods.h

```diff
@@ -0,0 +1,42 @@
+// cubic interpolation functions
+double interp1(double v1, double v2, double v3, double v4);
+double interp1(double *v);
+double interp2(double v1, double v2, double v3, double v4);
+double interp2(double *v);
+double interp3(double v1, double v2, double v3, double v4);
+double interp3(double *v);
+
+double cubic_interpolation(int interp_pos, double *v);
+
+// error checking (whether interpolation can be performed)
+
+void checkInterpolationPoints(int i_l, int i_u, int j_l, int j_u, int k_l, int k_u, int I, int J, int K);
+
+// bandlimited interpolation functions
+
+double bandlimited_interpolation(int interp_pos, double *a, int offset = 0);
+double bandlimited_interpolation(int interp_pos, double a0, double a1, double a2, double a3,
+                                 double a4, double a5, double a6, double a7);
+
+/**
+ * @brief Defines aliases for the flags that will determine the interpolation scheme to be used.
+ * 
+ * The choice of expansion is not arbitrary: for the band-limited cases, the expanded number can be used to tell us the range of Yee indicies we need to extract values from, to then interpolate.
+ * Similarly for cubic interpolation: the expanded number, offset by 7, fulfils the same role.
+ */
+enum interp_scheme
+{
```

Comment:
```suggestion
enum interp_scheme {
```
we can probably clang-format everything but probably good to be consistent with the rest of the codebase (here and it lots of other places 😅)

Reply:
Agreed - VSCode defaults to that style so I'll be vigilant from now on :sweat_smile: